esrally/client/factory.py

```diff
@@ -226,13 +220,26 @@ async def on_request_end(session, trace_config_ctx, params):
 
         async_client = RallyAsyncElasticsearch(
             distribution_version=self.distribution_version,
+            distribution_flavor=self.distribution_flavor,
             hosts=self.hosts,
             transport_class=RallyAsyncTransport,
             ssl_context=self.ssl_context,
             maxsize=self.max_connections,
             **self.client_options,
         )
 
+        async def on_request_start(session, trace_config_ctx, params):
+            async_client.on_request_start()
+
+        async def on_request_end(session, trace_config_ctx, params):
+            async_client.on_request_end()
+
+        trace_config = aiohttp.TraceConfig()
+        trace_config.on_request_start.append(on_request_start)
+        trace_config.on_request_end.append(on_request_end)
+        # ensure that we also stop the timer when a request "ends" with an exception (e.g. a timeout)
+        trace_config.on_request_exception.append(on_request_end)
+
```

Comment:
Would you mind reverting this change? (Move this back up and replace `async_client` with `RallyAsyncElasticsearch` in the async methods.)

Reply:
Addressed in https://github.com/elastic/rally/pull/1738/commits/94d863418f013b6d1a5498fb476f93b8cfb31707